### PR TITLE
SNOW-619385: Pass options set in session.read.options to df.copy_into_table

### PR DESCRIPTION
**Pull request by sfc-gh-yixie**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-619385

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/integ/scala/test_dataframe_copy_into.py [hunk 1]
```diff
@@ -309,11 +309,11 @@ def test_copy_csv_transformation(session, tmp_stage_name1, trans_columns):
             session.table(table_name),
             [
                 Row("1", "one", "1.2"),
-                Row("2", "two", "2.2"),
                 Row("1.2", "one", "1"),
+                Row("2", "two", "2.2"),
                 Row("2.2", "two", "2"),
             ],
-            sort=False,
+            sort=True,
```
Comment (sfc-gh-yixie) [inline tests/integ/scala/test_dataframe_copy_into.py:316]: This should be the reason for the test to be flaky.

File: src/snowflake/snowpark/_internal/utils.py [hunk 2]
```diff
@@ -115,6 +126,20 @@
     "LOAD_UNCERTAIN_FILES",
 }
 
+OTHER_NON_FORMAT_TYPE_OPTIONS = {
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/_internal/utils.py:129]: Maybe just `NON_FORMAT_TYPE_OPTIONS`? `OTHER` feels kinda weird.

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/_internal/utils.py (hunk 2 above)
Comment (sfc-gh-yixie) [reply src/snowflake/snowpark/_internal/utils.py]: Updated

File: src/snowflake/snowpark/_internal/utils.py [hunk 3]
```diff
@@ -115,6 +126,20 @@
     "LOAD_UNCERTAIN_FILES",
 }
 
+NON_FORMAT_TYPE_OPTIONS = {
+    "PATTERN",
+    "VALIDATION_MODE",
+    "FILE_FORMAT",
+    "FORMAT_NAME",
+    "FILES",
+    # The following are not copy into command options but client side options.
```
Comment (sfc-gh-jdu) [inline src/snowflake/snowpark/_internal/utils.py:135]:
```suggestion
    # The following are not copy into SQL command options but client side options.
```

**Review by sfc-gh-jdu — APPROVED** [review]

File: tests/integ/scala/test_dataframe_copy_into.py (hunk 1 above)
Comment (sfc-gh-jdu) [reply tests/integ/scala/test_dataframe_copy_into.py:316]: Would you like to remove @pytest.mark.xfail on this test to see whether it fixes the flaky tests?

Comment (sfc-gh-yixie) [reply tests/integ/scala/test_dataframe_copy_into.py:317]: Removed

**Pull request merged**